packages/wallet/src/redux/protocols/channel-management/reducer.ts

```diff
@@ -0,0 +1,84 @@
+import { SharedData } from '../../state';
+import { ProtocolStateWithSharedData } from '..';
+import * as states from './states';
+import { isChannelManagementAction } from './actions';
+import { unreachable } from '../../../utils/reducer-utils';
+import {
+  getOurAddress,
+  getOpponentAddress,
+  getOurAllocation,
+  getOpponentAllocation,
+  getOpenLedgerChannels,
+  showWallet,
+  hideWallet,
+  isTwoPlayerChannel,
+  getTotalAllocation as getAllocationTotal,
+  getOpenApplicationChannels,
+  getTargetOfLedgerFunding,
+  getFundingChannelId,
+} from '../reducer-helpers';
+import { WalletAction } from '../../actions';
+
+const mapToDisplayChannel = (channelId: string, sharedData: SharedData): states.DisplayChannel => {
+  const fundedBy = getFundingChannelId(channelId, sharedData);
+  const fundingChannel = getTargetOfLedgerFunding(channelId, sharedData);
+  return {
+    channelId,
+    ourAddress: getOurAddress(channelId, sharedData),
+    opponentAddress: getOpponentAddress(channelId, sharedData),
+    ourAmount: getOurAllocation(channelId, sharedData),
+    opponentAmount: getOpponentAllocation(channelId, sharedData),
+    channelAmount: getAllocationTotal(channelId, sharedData),
+    fundingChannel: fundingChannel === channelId ? '' : fundingChannel,
+    fundedBy,
+  };
+};
+export const initialize = ({
```

Comment:
I'm not sure if using a protocol is the best way to handle something that is mostly UI driven. It seems against the "engine" pattern we've been discussing.